docs/source/util.py

```diff
@@ -0,0 +1,174 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
+import argparse
+import os
+import shutil
+import subprocess
+from collections import namedtuple
+from typing import Iterable, Optional
+
+Class = namedtuple(
+    "Class",
+    ["module", "methods", "attributes"],
+    defaults=["snowflake.snowpark", [], []],
+)
+Module = namedtuple(
+    "Module", ["name", "attributes", "functions", "classes", "exceptions"]
+)
+mapping = {
+    "Methods": "methods",
+    "Attributes": "attributes",
+    "Module Attributes": "attributes",
+    "Functions": "functions",
+    "Classes": "classes",
+    "Exceptions": "exceptions",
+}
+
+
+def autogen_and_parse_for_info(module_name: str, class_name: Optional[str] = None):
+    if class_name:
+        res = Class(module_name, [], [])
+        name = f"{module_name}.{class_name}"
+    else:
+        res = Module(module_name, [], [], [], [])
+        name = module_name
+
+    if not os.path.exists("temp"):
```

Comment:
I wonder if we should check and make sure we start from a new directory before generating contents? I'm not sure if reusing the same directory will cause some unexpected lingering file when it is supposed to be deleted.